experimental/fast_lossless/fast_lossless.cc

```diff
@@ -545,14 +727,51 @@ struct From9To13Bits {
   using pixel_t = int16_t;
   using upixel_t = uint16_t;
 
+  static void PrepareForSimd(const uint8_t* nbits, const uint8_t* bits,
+                             size_t n, uint8_t* nbits_simd,
+                             uint8_t* bits_simd) {
+    assert(n <= 16);
+    memcpy(nbits_simd, nbits, 16);
+    memcpy(bits_simd, bits, 16);
+  }
+
   static void EncodeChunk(upixel_t* residuals, const PrefixCode& code,
                           BitWriter& output) {
+#if defined(FASTLL_ENABLE_NEON_INTRINSICS) && FASTLL_ENABLE_NEON_INTRINSICS
+    for (int i : {0, 8}) {
+      uint16_t bits[8];
+      uint16_t nbits[8];
+      uint16_t bits_huff[8];
+      uint16_t nbits_huff[8];
+      uint16_t token[8];
+      TokenizeNeon(residuals + i, token, nbits, bits);
+      HuffmanNeonUpTo13(token, code, nbits_huff, bits_huff);
+      StoreNeonUpTo14(nbits, bits, nbits_huff, bits_huff, output);
+    }
+    return;
+#endif
     GenericEncodeChunk(residuals, code, output);
   }
+
+  size_t NumSymbols(bool doing_ycocg) const {
```

Comment:
Is there a way to do this with fewer duplicates lines of code? Different things need to be done at different bit depths, but it feels like there might be some code duplication that can be avoided...

Reply:
I don't think the reduction of lines would be worth the increase in complexity/readability hit...